### ISSUE TITLE
Hot reloading code assumes user in container is root

### ISSUE THREAD

**Issue by Addono** [devspace-cloud/devspace#1086]
**What happened?**  
```
/devspace-restart-helper: line 16: can't create /devspace-pid: Permission denied
```
**What did you expect to happen instead?**  
That hot reloading works independent from the user set in my `Dockerfile`.

**How can we reproduce the bug?** (as minimally and precisely as possible)  
Create a container where you change the user to be non-root (e.g. let your Dockerfile contain `USER notroot`), then try to launch this container in dev mode with automatic reloading enabled.

**Local Environment:**  
- DevSpace Version: 4.12.2 (according to Brew)
- Operating System: mac
- Deployment method: devspace dev

**Kubernetes Cluster:**  
- Cloud Provider: DevSpace Cloud

**Anything else we need to know?**  
As a workaround I altered my `Dockerfile` to support setting the user as an ARG variable, which in DevSpace is then configured as root. This works for `devspace dev`, however 1) now my production `Dockerfile` is changed to suits the needs of my dev tool, 2) if I would use DevSpace to deploy, then the user is still set to root in production.
https://github.com/Addono/container-registry-proxy/commit/4afef532749d88773dbbc13568c7262ff2849d05

The first workaround which came to mind was duplicating my Dockerfile, but that's a huge DRY violation, so I prefer this other workaround.

/kind bug

**Comment by LukasGentele** [comment]
@Addono Thanks for opening this issue. Totally agree. We should improve this. 

One solution I can think of is besides injecting the `ADD` statement for the restart helper, we could also inject a `RUN mkdir /devspace && chown $USER /devspace && chmod 0700 /devspace` to the Dockerfile. Then, we should use the file `/devspace/pid` instead of `/devspace-id` and the container should be able to run with a non-root user even without having to change the Dockerfile manually. I think that should work and would be a pretty simple solution.

What do think? @Addono @FabianKramm

**Comment by FabianKramm** [comment]
In general, I don't see any issues in changing this, however switching the path of the restart helper script will render images built with an older devspace version (with inject restart helper enabled) incompatible with this change, so we have to support the old path as well (or force an image rebuild for older images) to avoid any breaking changes.

**Comment by LukasGentele** [comment]
Implemented as part of [v4.13](https://github.com/devspace-cloud/devspace/releases/tag/v4.13.0)
